lib/model/migrations/20230104-01-add-entity-indices.js

```diff
@@ -0,0 +1,34 @@
+// Copyright 2023 ODK Central Developers
+// See the NOTICE file at the top-level directory of this distribution and at
+// https://github.com/getodk/central-backend/blob/master/NOTICE.
+// This file is part of ODK Central. It is subject to the license terms in
+// the LICENSE file found in the top-level directory of this distribution and at
+// https://www.apache.org/licenses/LICENSE-2.0. No part of ODK Central,
+// including this file, may be copied, modified, propagated, or distributed
+// except according to the terms contained in the LICENSE file.
+
+const up = async (db) => {
+  await db.schema.table('entities', (en) => {
+    en.index('datasetId');
+  });
```

Comment:
For the entities OData feed / the entities table in Frontend, are we going to sort by `"createdAt" desc, id desc` like we do for submissions? If so, then maybe we should also add an index involving `createdAt`? It looks like we did that for the `submissions` table:

https://github.com/getodk/central-backend/blob/9d767ff84ffc8fef1fa36fd97a723139091bb72a/lib/model/migrations/20181012-01-add-submissions-createdat-index.js#L10-L12

(To be clear, this would be in addition to the index you've added here.)

Definitely feel free to skip for now and circle back to this if/when it's clear whether we'll need an index on `createdAt`.

Reply:
I have updated the index to include `createdAt` and `id`, one index is sufficient to handle all of our use cases.